src/com/google/javascript/jscomp/PeepholeFoldConstants.java

```diff
@@ -1080,6 +1081,15 @@ static TernaryValue evaluateComparison(int op, Node left, Node right, boolean us
       case Token.NUMBER:
         if (right.isNumber()) {
           return compareAsNumbers(op, left, right, useTypes);
+        } else if (right.isString()) {
+          return compareAsNumbers(op, left, IR.number(NodeUtil.getNumberValue(right)), useTypes);
```

Comment:
This branch is not correct for SHEQ and SHNE, since, e.g. `'1' !== 1`.
I think that those cases are handled are handled earlier so that this branch isn't reachable for SHEQ and SHNE, but I'd rather that the code be more explicit about what cases are covered and what the ES semantics are. Links into the relevant parts of the spec would also be helpful.